src/wp-includes/class-wp-scripts.php

```diff
@@ -915,21 +970,20 @@ private function has_only_deferrable_dependents( $handle, $checked = array() ) {
 			return true;
 		}
 
-		// Consider each dependent and check if it is deferrable.
+		// Consider each dependent and check if it is delayed.
 		foreach ( $dependents as $dependent ) {
-			// If the dependent script is not using the defer or async strategy, no script in the chain is deferrable.
+			// If the dependent script is not using the defer or async strategy, no script in the chain is delayed.
 			$strategy = $this->get_data( $dependent, 'strategy' );
-			if ( ! $this->is_delayed_strategy( $strategy ) ) {
-				return false;
-			}
-
-			// If the dependent script has a non-standalone inline script in the 'before' position associated with it, do not defer.
-			if ( $this->has_non_standalone_inline_script( $dependent, 'before' ) ) {
+			if ( $async_only ) {
+				if ( 'async' !== $strategy ) {
+					return false;
+				}
+			} elseif ( ! $this->is_delayed_strategy( $strategy ) ) {
 				return false;
 			}
 
 			// Recursively check all dependents.
-			if ( ! $this->has_only_deferrable_dependents( $dependent, $checked ) ) {
+			if ( ! $this->has_only_delayed_dependents( $dependent, $async_only, $checked ) ) {
```

Comment:
Was curious about this method and discovered an existing problem that should likely be addressed. The `has_only_delayed_dependents` method relies on the newly introduced `get_dependents` method that returns all "registered" dependents, but we should only be concerned about "enqueued" scripts, not ones that are not going to be printed to the page.

We can address that separate from this PR, in order to not block this though.

Reply:
Good catch. Filed this as https://github.com/10up/wordpress-develop/issues/71